examples_tests/61.BlitFilterTest/normalization/descriptors.glsl

```diff
@@ -0,0 +1,55 @@
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTORS_INCLUDED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTORS_INCLUDED_
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_ 0
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_ nbl_glsl_blit_normalization_inOutImage
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_BINDING_DEFINED_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_BINDING_DEFINED_ 0
+	#endif
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_
+		#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ must be defined to any of (u)image1D/(u)image2D/(u)image3D
+	#endif
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_
+		#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_ must be defined to a valid storage image format
+	#endif
+
+	layout(set = _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_, binding = _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_BINDING_DEFINED_, _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_) uniform _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_;
+
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_ALPHA_HISTOGRAM_DESCRIPTOR_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_ALPHA_HISTOGRAM_DESCRIPTOR_DEFINED_ nbl_glsl_blit_normalization_alphaHistogram
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_ALPHA_HISTOGRAM_BINDING_DEFINED_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_ALPHA_HISTOGRAM_BINDING_DEFINED_ 1
+	#endif
+
+	layout(set = _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_, binding = _NBL_GLSL_BLIT_NORMALIZATION_ALPHA_HISTOGRAM_BINDING_DEFINED_) buffer restrict readonly AlphaHistogram
+	{
+		uint data[];
+	} _NBL_GLSL_BLIT_NORMALIZATION_ALPHA_HISTOGRAM_DESCRIPTOR_DEFINED_;
+
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_PASSED_INPUT_PIXEL_COUNT_DESCRIPTOR_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_PASSED_INPUT_PIXEL_COUNT_DESCRIPTOR_DEFINED_ nbl_glsl_blit_normalization_passedInputPixelCount
+
+	#ifndef _NBL_GLSL_BLIT_NORMALIZATION_PASSED_INPUT_PIXEL_COUNT_BINDING_DEFINED_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_PASSED_INPUT_PIXEL_COUNT_BINDING_DEFINED_ 2
+	#endif
+
+	layout(set = _NBL_GLSL_BLIT_NORMALIZATION_DESCRIPTOR_SET_DEFINED_, binding = _NBL_GLSL_BLIT_NORMALIZATION_PASSED_INPUT_PIXEL_COUNT_BINDING_DEFINED_) buffer restrict readonly PassedInputPixelCount
+	{
+		uint data;
+	} _NBL_GLSL_BLIT_NORMALIZATION_PASSED_INPUT_PIXEL_COUNT_DESCRIPTOR_DEFINED_;
+
+#endif
```

Comment:
oh btw, you don't need this SSBO at all, you can infer the value by querying the value of the cumulative histogram at the index matching the reference